docs/_content/ontwikkelaars/algemeen/audit-trail.md

```diff
@@ -0,0 +1,248 @@
+
+Bron: [NORA](https://www.noraonline.nl/wiki/Inhoud_audit_trail)
+
+We laten de relatie met logging (in het algemeen) en doelbinding volgens de AVG
+buiten beschouwing. Hier zijn andere oplossingen voor in de maak, zoals 
+[NLx](https://nlx.io).
+
+## API implementatie
+
+### Endpoint
+
+De API wordt gerealiseerd als alleen-lezen endpoint binnen de API van het 
+component, waarbij de resource "audittrail" wordt ontsloten op het gelijknamige
+endpoint. Voor het ZRC zal dit dus zijn: `/api/v1/audittrail`.
+
+De autit trail resource *kan* ingeladen worden in de OAS van het component 
+middels een referentie maar mag ook inline worden opgenomen in de OAS van het 
+component zelf.
+
+**Rationale**
+
+De audit trail API...
+
+1. ...wordt beschikbaar in een zelfstandig component worden waar alle andere 
+   componenten hun audit regels naar toe sturen.
+2. ...wordt onderdeel van de component API.
+
+We realiseren geen apart component voor de audit trail waar alle componenten
+hun logging heen sturen. Dit zou een synchrone afhankelijkheid creëeren en er
+gaat potentieel veel informatie over het netwerk wat veel overhead veroorzaakt,
+zelfs als niemand deze audit trail ooit raad pleegt. Ten slotte staat deze 
+opzet een eventueel centraal logging component niet in de weg.
+
+De audit trail API wordt dus onderdeel van elk component. Er zijn 2 keuzes, de 
+audit trail API...
+
+1. ...krijgt een zelfstandige API naast de component API 
+   (`example.com/zrc/api/v1` en `example.com/zrc/audittrail/api/v1`).
+2. ...wordt onderdeel van de component API 
+   (`example.com/zrc/api/v1/audittrail`).
+
+Bij het NRC is voor het ontvangen van notificaties door componenten gekozen 
+voor een zelfstandige API. Dit voorkomt dat een wijziging voor de audit trail 
+API alle API's van componenten raakt. Echter, bij het NRC geeft de consumer 
+zelf aan waar deze API te vinden is. Bij de audit trail is hiervan geen sprake
+en dus moet de API consistent en snel te vinden zijn, en het is in essentie ook
+echt een onderdeel van het component.
+
+### Attributen
+
+`uuid` (automatisch)
+
+Unieke identificatie van de audit regel.
+
+`bron` (automatisch)
+
+De naam van get component waar de wijziging in is gedaan. Hoewel dit attribuut 
+in de context van de API niet relevant lijkt, aangezien de API onderdeel is van 
+het component. Mocht deze API echter gebruikt worden in een geaggregeerde 
+context, dan blijft het formaat voor consumers gelijk.
+
+`applicatieId` (verplicht)
+
+Unieke identificatie van de applicatie, binnen de organisatie. Deze komt 
+overeen met het `client_id` attribuut uit de JWT.
+
+`applicatieWeergave` (optioneel)
+
+Vriendelijke naam van de applicatie. Deze kan worden afgeleid uit het `label` 
+attribuut van de `Applicatie` resource in het Atorisatie Component (AC).
+
+`gebruikersId` (verplicht) 
+
+Unieke identificatie van de gebruiker, binnen de organisatie. Deze moet uit de 
+JWT worden gehaald.
```

Comment:
De unieke identificatie van de gebruiker binnen de organisatie stelt meteen eisen aan de taakapplicatie: die moet immers beschikken over deze identificatie. De vraag is of huidige taakapplicaties hier nu aan (kunnen) voldoen. Veel applicaties zullen als ze op het AD aangesloten zijn, het SAMaccountname wel hebben of een email adres, maar dat hoef niet. 
Beter is daarom misschien: de unieke identificatie van de gebruiker zoals die binnen de applicatie bekend is. 
Herleidbaarheid tot een natuurlijk persoon kan dan bereikt worden door de combinatie van de applicatieId + gebruikersId

Reply:
Goede opmerking maar ik zou het wel willen omdraaien, zodat er komt te staan dat we het **liefst** de globale `gebruikerId` hebben (zoals uit het AD) maar dat je daar ook de identificatie binnen een applicatie mag gebruiken. Volgens mij is de eerste namelijk makkelijker te herleiden dan de laatste, maar moet het beide mogen. Eens of niet?

Reply:
Ik ben het op zich met je eens. Je wilt zoveel mogelijk de globale ID hebben, zolang de gebruiker ook maar onder dit ID in de taakapplicatie te vinden is.
Punt van aandacht is overigens wel: het zou kunnen dat organisaties het personeelsnummer gebruiken als unieke identificatie (wat mij betreft wel onverstandig). Het personeelsnummer is een persoonsgegeven dat je goed wilt beschermen.  Daar zal je dan heel goed rekening mee moeten houden als je mensen toegang geeft tot de API...

Reply:
Ik heb er van gemaakt:

> Unieke identificatie van de gebruiker die binnen de organisatie herleid kan
worden naar een persoon. Dit mag bijvoorbeeld het ID zijn van de gebruiker
zoals deze bekend is in Active Directory, of het ID van de gebruiker zoals deze
alleen binnen de applicatie bekend is. In het laatste geval kan de persoon met 
behulp van het `applicatieId` ook tot een persoon herleid worden.